src/blocks/AppsProcessor.cpp

```diff
@@ -1,20 +1,94 @@
 #include "AppsProcessor.h"
+#include "util/filters/Timer.h"
 
 /**
  * Take APPS raw analog voltages and convert them to pedal travel percentages.
  * Make sure the two sensors are both within range and in agreement with each other.
  *
  * Read the rules!
  */
-
+Timer clock(0.1);
 void AppsProcessor::evaluate(VcuParameters* params, AppsProcessorInput* input,
                              AppsProcessorOutput* output, float deltaTime) {
 
-    output->apps = (input->apps1 + input->apps2) / 2.0f; // do not use lmao
-    output->ok = false;
 
-    // TODO use parameters
-    // TODO implement plausibility
+    /**
+     *
+     *
+     *
+     */
+    //check if inputs are within working range first
+
+    //checking whether app1 input is greater than max voltage params or less than min voltage params
+    if(input->apps1 < params->apps1VoltageMin)
+    {
+        output->ok = false;
+        return;
+    } else if (input->apps1 > params->apps1VoltageMax)
+    {
+        output->ok = false;
+        return;
+    }
+    //checking whether app2 input is greater than max voltage params or less than min voltage params
+    if(input->apps2 < params->apps2VoltageMin)
+    {
+        output->ok = false;
+        return;
+    } else if (input->apps2 > params->apps2VoltageMax)
+    {
+        output->ok = false;
+        return;
+    }
+    // check if we need evaluation or not
+    
+    // since percentage will be (input - min) / (max - min)
+    // finding max - min
+    float diff = params->apps1VoltageMax - params->apps1VoltageMin;
+    // finding input - min
+    float app1Perc = input->apps1 - params->apps1VoltageMin;
+    float app2Perc = input->apps2 - params->apps2VoltageMin;
+    // calc percentage
+    app1Perc = app1Perc / diff;
+    diff = params->apps2VoltageMax - params->apps2VoltageMin;
+    app2Perc = app2Perc / diff;
```

Comment:
compute the max-min difference on the spot instead of assign "diff" because that assigns multiple different uses to the "diff" variable and can be confusing/strange to debug